quartz_solar_forecast/forecasts/v2.py

```diff
@@ -0,0 +1,226 @@
+
+        Parameters
+        ----------
+        latitude : float
+            Latitude of the location.
+        longitude : float
+            Longitude of the location.
+        start_date : str
+            Start date in 'YYYY-MM-DD' format.
+        kwp : float
+            Kilowatt peak of the solar panel system.
+        orientation : float
+            Orientation angle of the solar panel system in degrees.
+        tilt : float
+            Tilt angle of the solar panel system in degrees.
+
+        Returns
+        -------
+        pd.DataFrame
+            Prepared weather data with additional solar panel parameters.
+        """
+        start_date_datetime = datetime.datetime.strptime(start_date, "%Y-%m-%d")
+        end_date_datetime = start_date_datetime + datetime.timedelta(days=2)
+        end_date = end_date_datetime.strftime("%Y-%m-%d")
+
+        weather_service = WeatherService()
+
+        weather_data = weather_service.get_15_minutely_weather(
+            latitude, longitude, start_date, end_date
+        )
+
+        PANEL_COLUMNS = [
+            "latitude_rounded",
+            "longitude_rounded",
+            "orientation",
+            "tilt",
+            "kwp",
+        ]
+
+        weather_data["latitude_rounded"] = latitude
+        weather_data["longitude_rounded"] = longitude
+        weather_data["orientation"] = orientation
+        weather_data["tilt"] = tilt
+        weather_data["kwp"] = kwp
+
+        cols = PANEL_COLUMNS + [
+            col for col in weather_data.columns if col not in PANEL_COLUMNS
+        ]
+        weather_data = weather_data[cols]
+
+        return weather_data
+
+    def clean(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Cleans and transforms the input DataFrame.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame containing the weather and solar panel data.
+
+        Returns
+        -------
+        pd.DataFrame
+            Transformed DataFrame ready for prediction.
+        """
+        df.loc[:, self.DATE_COLUMN] = pd.to_datetime(df[self.DATE_COLUMN])
+
+        df.loc[:, "date_year"] = df[self.DATE_COLUMN].dt.year
+        df.loc[:, "date_month"] = df[self.DATE_COLUMN].dt.month
+        df.loc[:, "date_day"] = df[self.DATE_COLUMN].dt.day
+        df.loc[:, "date_hour"] = df[self.DATE_COLUMN].dt.hour
+        df.loc[:, "date_minute"] = df[self.DATE_COLUMN].dt.minute
+
+        COLUMNS_TO_DROP = ["date_minute", "date_year",
+                           "terrestrial_radiation",
+                           "shortwave_radiation",
+                           "direct_normal_irradiance"]
+
+        df = df.drop(columns=COLUMNS_TO_DROP)
+
+        return df
+
+    def predict_power_output(
+        self,
+        latitude: float,
+        longitude: float,
+        start_date: str,
+        kwp: float,
+        orientation: float,
```

Comment:
default values here too?

Reply:
Added the same default values as before.